tnefparse/tnef.py

```diff
@@ -402,8 +401,7 @@ def to_zip(data, default_name='no-name', deflate=True):
         L = len(tozip.get(filename, []))
         if L > 0:
             # uniqify this file name by adding -<num> before the extension
-            root, ext = os.path.splitext(filename)
-            tozip[filename].append((attachment.data, str("%s-%d%s" % (root, L + 1, ext))))
+            tozip[filename].append((attachment.data, filename.with_stem(f"{filename.stem}-{L + 1}")))
```

Comment:
As this line is not covered by a test, this would be a great chance to do so :-)

Reply:
Added an extra test.